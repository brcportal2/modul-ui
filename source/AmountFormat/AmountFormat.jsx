import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {validateHelper} from 'modul-helpers';
import accounting from 'accounting';


const CurrencySymbol = ({value}) => {
	switch (value) {
		case 'USD':
			return <span className='cur dollar'><span>$</span></span>;
		case 'EUR':
			return <span className='cur euro'><span>€</span></span>;
		default:
			return <span className='cur ruble'><span>р.</span></span>;
	}
};

export default class AmountFormat extends Component {
	static displayName = 'AmountFormat';

	static propTypes = {
		value: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number
		]),
		currency: PropTypes.oneOf([
			'RUR',
			'USD',
			'EUR'
		]),
		precision: PropTypes.number,
		def: PropTypes.string,
		className: PropTypes.string
	};

	static defaultProps = {
		currency: 'RUR',
		precision: 2,
		def: '',
		className: ''
	};

	clear = value => {
		return value.replace
			? value.replace(/[^0-9.,]+/g, '').replace(',', '.')
			: value;
	};

	render() {
		const {
			value,
			def,
			currency,
			precision,
			...props
		} = this.props;

		if (validateHelper.isEmpty(value))
			return def
				? <span {...props}>{def}</span>
				: null;

		const val = parseFloat(this.clear(value));
		const formatted = !isNaN(val)
			? accounting.formatNumber(val, precision, ' ')
			: def;

		return (
			<span {...props}>
				{formatted} &nbsp; <CurrencySymbol value={currency} />
			</span>
		);
	}
}