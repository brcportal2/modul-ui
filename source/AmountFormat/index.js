import AmountFormat from './AmountFormat';


export default AmountFormat;
export {
    AmountFormat
};