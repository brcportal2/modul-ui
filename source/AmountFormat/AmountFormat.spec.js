import React from 'react';
import renderer from 'react-test-renderer';

import AmountFormat from './AmountFormat';


describe('AmountFormat', () => {
	test('Default props', () => {
		const component = renderer.create(
			<AmountFormat
				value='123456'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Default value', () => {
		const component = renderer.create(
			<AmountFormat
				def='not date'
				value='123456'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('ClassName', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='123456'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Big value', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='1234567890'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Currency RUR', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='1234567890'
				currency='RUR'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Currency USD', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='1234567890'
				currency='USD'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Currency EUR', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='1234567890'
				currency='EUR'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('First zero', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value='0123'
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test('Null value', () => {
		const component = renderer.create(
			<AmountFormat
				className='test_class'
				def='not date'
				value={null}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});