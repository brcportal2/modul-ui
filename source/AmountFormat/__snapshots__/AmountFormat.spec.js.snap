// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountFormat Big value 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "1&nbsp;234&nbsp;567&nbsp;890.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat ClassName 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "123&nbsp;456.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat Currency EUR 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "1&nbsp;234&nbsp;567&nbsp;890.00",
      }
    }
  />
   
  <span
    className="cur euro"
  >
    <span>
      €
    </span>
  </span>
</span>
`;

exports[`AmountFormat Currency RUR 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "1&nbsp;234&nbsp;567&nbsp;890.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat Currency USD 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "1&nbsp;234&nbsp;567&nbsp;890.00",
      }
    }
  />
   
  <span
    className="cur dollar"
  >
    <span>
      $
    </span>
  </span>
</span>
`;

exports[`AmountFormat Default props 1`] = `
<span
  className=""
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "123&nbsp;456.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat Default value 1`] = `
<span
  className=""
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "123&nbsp;456.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat First zero 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "123.00",
      }
    }
  />
   
  <span
    className="cur ruble"
  >
    <span>
      р.
    </span>
  </span>
</span>
`;

exports[`AmountFormat Null currency 1`] = `
<span
  className="test_class"
>
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "123.00",
      }
    }
  />
   
</span>
`;

exports[`AmountFormat Null value 1`] = `
<span
  className="test_class"
>
  not date
</span>
`;
